apollo-federation/src/sources/connect/models/validation/mod.rs

```diff
@@ -370,32 +385,13 @@ fn validate_field(
         })
         .collect();
 
-    if http_methods.len() > 1 {
-        errors.push(Message {
-            code: Code::MultipleHttpMethods,
-            message: format!(
-                "{coordinate} cannot specify more than one HTTP method.",
-                coordinate =
-                    connect_directive_http_coordinate(connect_directive_name, object, &field.name),
-            ),
-            locations: http_methods
-                .iter()
-                .flat_map(|(_, node)| Location::from_node(node.location(), source_map).into_iter())
-                .collect(),
-        });
-    } else if http_methods.is_empty() {
-        errors.push(Message {
-            code: Code::MissingHttpMethod,
-            message: format!(
-                "{coordinate} must specify an HTTP method.",
-                coordinate =
-                    connect_directive_http_coordinate(connect_directive_name, object, &field.name),
-            ),
-            locations: Location::from_node(http_arg_location, source_map)
-                .into_iter()
-                .collect(),
-        });
-    }
+    errors.extend(validate_http_method(
```

Comment:
The code above to gather the HTTP methods seems like it should also be moved into the `http_method` module. Just the fact that this takes in an array of HTTP methods is tied to the logic that validates whether more than one is specified. Separating that logic across two files introduces an unnecessary interdependency there.